src/ApiController.php

```diff
@@ -0,0 +1,284 @@
+     *
+     * @return closure outputs a single emoji in json format
+     */
+    public function fetchEmoji()
+    {
+        return function ($id) {
+            //set the response header content-type and fetch emoji
+            $this->app->response->headers->set('Content-Type', 'application/json');
+            $emoji = $this->db->emoji('id = ?', $id)->fetch();
+
+            //if emoji not found return error message else return emoji
+            if ($emoji == null) {
+                $this->app->response->setStatus(204);
+                echo json_encode(['message' => 'Emoji not found']);
+            } else {
+                echo json_encode($emoji);
+            }
+        };
+    }
+
+    /**
+     * Controller action that processes the request made to the create emoji endpoint.
+     *
+     * @return closure outputs a status message and emoji created in json format
+     */
+    public function createEmoji()
+    {
+        return function () {
+            //set response header content-type and retrieve user token
+            $this->app->response->headers->set('Content-Type', 'application/json');
+            $token = $this->app->request->headers->get('user-token');
+
+            //initialize the emoji db model
+            $emoji = $this->db->emoji();
+
+            //check if token is valid then proceed or return an error message
+            if (Token::isValid($token, $this->db)) {
+                //get post data and make some additions
+                $all_post_vars = $this->app->request->post();
+                $all_post_vars['date_created'] = $this->get_current_time('Y-m-d H:i:s');
+                $all_post_vars['date_modified'] = $this->get_current_time('Y-m-d H:i:s');
+                $all_post_vars['created_by'] = Token::owner($token, $this->db);
+
+                //insert record
+                $res = $emoji->insert($all_post_vars);
+                $timestamp = $res['date_created'];
+
+                //return result of this request operation and the inserted emoji
+                echo json_encode([
+                    'message' => 'Emoji successfully added!',
+                    'emoji'   => $emoji->where('date_created = ?', $timestamp)->fetch(),
+                    ]);
+            } else {
+                $this->app->response->setStatus(401);
+                echo json_encode(['message' => 'Invalid Token!']);
+            }
+        };
+    }
+
+    /**
+     * Controller action that processes the request made to the update endpoint via either put or patch.
+     *
+     * @return closure outputs a status message 
+     */
+    public function updateEmoji()
+    {
+        return function ($id) {
+            $this->app->response->headers->set('Content-Type', 'application/json');
+            $token = $this->app->request->headers->get('user-token');
+
+            if (Token::isValid($token, $this->db) && $this->user_owns_emoji($token, $id) && $this->emoji_exists($id)) {
```

Comment:
**Code Quality - Nitpick**
This is long, split it into multi-line following [this guideline](https://pear.php.net/manual/en/rfc.cs-enhancements.splitlongstatements.php).